src/main/java/seedu/duke/common/Utilities.java

```diff
@@ -0,0 +1,122 @@
+package seedu.duke.common;
+
+import seedu.duke.exception.InputTransactionUnknownTypeException;
+import seedu.duke.exception.InputTransactionExtraTagException;
+import seedu.duke.exception.InputTransactionInvalidCategoryException;
+import seedu.duke.exception.InputTransactionInvalidDateException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static seedu.duke.common.DateFormats.DATE_INPUT_PATTERN;
+
+/**
+ * Provides a set of extra utilities for processing of different inputs in various commands.
+ */
+public class Utilities {
```

Comment:
Since we are extracting the methods used for parsing tags from AddCommand out so that it can be used for other commands, I think a better naming for it is to call it a "TagParser" or "ParameterParser" inside the Parser package. In this case, we will have a CommandParser (the current Parser.java) and a ParameterParser to parse the whole commands entered by the user into "commands" and "parameters".

I will help to do this later since I need to make some changes on command parsing later too to move all the parsing methods used in the command package into the parser package, so that our code structure follows the Separation of Concern principle.

Reply:
Sounds good, so that it's more clearly defined for usage by different commands. Ok you will work on this later.